pkg/plugin/conntrack/_cprog/conntrack.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "vmlinux.h"
+#include "bpf_helpers.h"
+
+// Helper functions to get the current time
+// Ref: https://github.com/cilium/cilium/blob/6186d579ed60f334c7a4daaf81060797b02cc6bd/bpf/lib/time.h
+#define NSEC_PER_SEC	(1000ULL * 1000ULL * 1000UL)
+#define bpf_ktime_get_sec()	\
+	({ __u64 __x = bpf_ktime_get_boot_ns() / NSEC_PER_SEC; __x; })
+# define bpf_mono_now()		bpf_ktime_get_sec()
+
+// These units are in seconds
+
+// Define how long a TCP connection should be kept in the table
+#define CT_CONNECTION_LIFETIME_TCP 360 
+// Define how long a non-TCP connection should be kept in the table
+#define CT_CONNECTION_LIFETIME_NONTCP 60
+// Define how long a TCP connection should be kept alive after receiving the first SYN
+#define CT_SYN_TIMEOUT 60
+// Define the interval at which a packet should be sent to the userspace
+#define CT_REPORT_INTERVAL 30
+// Define the maximum number of connections that can be stored in the conntrack table
+#define CT_MAP_SIZE 262144
+
+enum tcp_flags {
+    TCP_FIN = 0x01,
+    TCP_SYN = 0x02,
+    TCP_RST = 0x04,
+    TCP_PSH = 0x08,
+    TCP_ACK = 0x10,
+    TCP_URG = 0x20,
+    TCP_ECE = 0x40,
+    TCP_CWR = 0x80,
+    TCP_NS  = 0x100
+};
+
+enum ct_packet_dir {
+    CT_FORWARD = 0,
+    CT_REPLY,
+};
+
+enum ct_traffic_dir {
```

Comment:
I strongly advise against introducing new enums like these - a single byte is enough.
Sadly, unlike in C++ where you can specify the underlying type, in C we can't really :(
Enum values are great for readability but cause terrible struct bloat and the ebpf programs on in the hot path of our dataplane.

Reply:
yea i tried looking up and see if we could define the type for enum in C since it is a whopping 4 bytes by default and came to the same result as well (Copilot lied about this btw 🥲). Will change this to a byte and then add comments for clarification.

Reply:
Can we `#define` some bytes instead? Not as pretty and lacking guarantees, but it's at least a little more self-documenting

```c
#define TRAFFIC_DIRECTION_UNKNOWN 0x00
#define TRAFFIC_DIRECTION_INGRESS 0x01
#define TRAFFIC_DIRECTION_EGRESS 0x02